app/store/GeoJsonStore.js

```diff
@@ -104,23 +107,28 @@ class GeoJsonStore extends Store {
     }
     if (!this.geoJsonData[id]) {
       this.geoJsonData[id] = 'pending';
-      const responses = await Promise.all(urlArr.map(u => getJson(u)));
-      const mapped = responses.map(r => {
-        if (metadata) {
-          MapJSON(r, metadata);
+      try {
+        const responses = await Promise.all(urlArr.map(u => getJson(u)));
+        const mapped = responses.map(r => {
+          if (metadata) {
+            MapJSON(r, metadata);
+          }
+          return styleFeatures(r);
+        });
+        for (let i = 1; i < mapped.length; i++) {
+          mapped[0].features = mapped[0].features.concat(mapped[i].features);
         }
```

Comment:
Could these loops be combined with something like
```javascript
  const features = [];
  const mapped = responses.map(r => {
    if (metadata) {
      MapJSON(r, metadata);
    }
    const styledFeature = styleFeatures(r);
    features.push(...styledFeature.features);
    return styledFeature;
  });

  mapped[0].features = features;
```

Reply:
ok, I will refactor

Reply:
logic is now much easier to understand

Reply:
Even better! 👏